module-apps/application-phonebook/windows/PhonebookMultipleNumbersSelect.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "PhonebookMultipleNumbersSelect.hpp"
+
+#include <application-phonebook/ApplicationPhonebook.hpp>
+#include <application-phonebook/data/PhonebookItemData.hpp>
+
+namespace gui
+{
+    PhonebookMultipleNumbersSelect::PhonebookMultipleNumbersSelect(app::ApplicationCommon *app,
+                                                                   std::shared_ptr<MultipleNumbersModel> &&numberModel)
+        : AppWindow(app, gui::window::name::multiple_numbers_select), numberModel(std::move(numberModel))
+    {
+        buildInterface();
+    }
+
+    void PhonebookMultipleNumbersSelect::buildInterface()
+    {
+        AppWindow::buildInterface();
+
+        numberList = new gui::ListView(this,
+                                       phonebookStyle::mainWindow::contactsList::x,
+                                       phonebookStyle::mainWindow::contactsList::y,
+                                       phonebookStyle::mainWindow::contactsList::w,
+                                       phonebookStyle::mainWindow::contactsList::h,
+                                       numberModel);
+        numberList->setBoundaries(Boundaries::Continuous);
+        setFocusItem(numberList);
+
+        navBar->setText(nav_bar::Side::Center, utils::translate(style::strings::common::select));
+        navBar->setText(nav_bar::Side::Right, utils::translate(style::strings::common::back));
+
+        setFocusItem(numberList);
+    }
+
+    void PhonebookMultipleNumbersSelect::onBeforeShow(ShowMode mode, SwitchData *data)
+    {
+        auto numberSelectData = dynamic_cast<PhonebookMultipleNumbersRequest *>(data);
+        assert(numberSelectData);
```

Comment:
tbh assert won;t be evaluated in release build, so you only have the check in debug builds, am I wrong?

Reply:
Hmmm no idea to be honest -> we are using asserts everywhere :) that is a good question

Reply:
As discussed in that case assert is ok as that is situation that should never happen.